lib/lab_manager/provider/vsphere.rb

```diff
@@ -105,7 +108,20 @@ def create_vm(opts = {})
     end
 
     def terminate_vm(_opts = {})
-      VSphere.connect.with do |_vs|
+      fail ArgumentError, 'Virtual machine data not present' unless instance_uuid
+
+      VSphere.connect.with do |vs|
+        Retryable.retryable(tries: 3) do
+          unless vs.get_virtual_machine(instance_uuid)['power_state'] == 'poweredOff'
+            vs.vm_power_off('force' => true, 'instance_uuid' => instance_uuid)
+          end
```

Comment:
fyi: there is another way how to terminate machine: https://github.com/CenturyLinkCloud/chef-provisioning-vsphere/blob/5c9bbba9999de7e0e58f99cccb3b365d7617f413/lib/chef/provisioning/vsphere_driver/vsphere_helpers.rb#L57-L69

it uses `.ShutdownGuest` which is the same case as if you set ` 'force' => false` in fog